src/behavior.js

```diff
@@ -224,14 +226,6 @@ export default class Behavior extends React.PureComponent {
       return { ...obj, [key]: rest[key] }
     }, {})
 
-    if (freeze) {
-      return (
-        <View pointerEvents={pointerEvents} ref={this.ref} style={[style, viewStyles, propStyles]}>
-          {children}
-        </View>
-      )
-    }
```

Comment:
By moving this down you execute extra math you don't need for `nativeStyles`, did you find that we actually need them?

Reply:
I am also taking into consideration on extra math, and yes we cant avoid the extra math on generating the `inputRange`

However, I have two reasons why I finally decided taking this approach:
1. we still managed to avoid the extra math on `getRange` functions due to [this line](https://github.com/ExodusMovement/haraka/blob/nafis/disable-any-animation-on-freeze/src/behavior.js#L259)
2. on freeze mode, we still need to translate `state` into similar object structure with the non-freeze mode, hence we can avoid having two similar logic exists on the code. For example we have the following components defined on UI:
```
<Behavior currentState={0} initialState={0} state=[{opacity: 0.5, translateY: 10},{opacity: 1, translateY: 100}] ...
```

On freeze mode it will be translated to:
```
... style={{ 
  opacity: 0.5, 
  transform: { translateY: 10 } 
}} ...
```

And, on non-freeze mode:
```
... style={{ 
  opacity: *Animation.interpolation object*, 
  transform: { translateY: *Animation.interpolation object* } 
}} ...
```